addons/hr_timesheet/models/hr_employee.py

```diff
@@ -24,3 +26,12 @@ def action_unlink_wizard(self):
             'target': 'new',
             'context': self.env.context,
         }
+
+    def action_timesheet_from_employee(self):
+        action = self.env["ir.actions.act_window"]._for_xml_id("hr_timesheet.timesheet_action_from_employee")
+        context = literal_eval(action['context'].replace('active_id', str(self.id)))
+        context.update({
+            'create': context.get('create', True) and self.active,
+        })
```

Comment:
We could directly do this since we just want to update one key. 🙂 
```suggestion
        context['create'] = context.get('create', True) and self.active
```

Reply:
don't forget to update your commit message since the user can edit. 🙂